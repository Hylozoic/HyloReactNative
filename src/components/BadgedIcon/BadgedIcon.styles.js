import { StyleSheet } from 'react-native'
import { persimmon } from '../../style/colors'

export default StyleSheet.create({
  badgeInner: {
    backgroundColor: persimmon,
    borderRadius: 4,
    height: 8,
    width: 8,
  },
  badgeOuter: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'white',
    borderRadius: 5.5,
    position: 'absolute',
    height: 11,
    width: 11,
    top: 1,
    right: 16
  }
})
