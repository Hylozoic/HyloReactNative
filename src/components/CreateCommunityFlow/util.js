export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'
