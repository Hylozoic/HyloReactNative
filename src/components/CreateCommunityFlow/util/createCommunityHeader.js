import React from 'react'
import {
  Text
} from 'react-native'
import { HeaderBackButton } from 'react-navigation'
import { caribbeanGreen, white, white60onCaribbeanGreen } from 'style/colors'

export default function createCommunityHeader (title, navigation) {
  return {
    title: <Text style={styles.headerTitle}>{title}</Text>,
    color: white,
    fontSize: 10,
    headerStyle: {
      backgroundColor: caribbeanGreen,
      borderBottomWidth: 0,
      shadowRadius: 0,
      shadowOffset: {
        height: 0
      },
      elevation: 0
    },
    headerBackgroundColor: caribbeanGreen,
    headerLeft: <HeaderBackButton onPress={() => navigation.goBack()} tintColor={white60onCaribbeanGreen} />
  }
}

const styles = {
  headerTitle: {
    color: white,
    fontSize: 12,
    fontWeight: 'bold'
  }
}
