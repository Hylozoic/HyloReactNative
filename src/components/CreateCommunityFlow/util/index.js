export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (communityDomain) {
  if (!communityDomain) return
  let formattedDomain = communityDomain.replace(/hylo\.com\/c\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/c/' + formattedDomain
  }
  return formattedDomain
}

export function removeUrlFromDomain (communityDomain) {
  if (!communityDomain) return
  return communityDomain.replace('hylo.com/c/', '')
}
