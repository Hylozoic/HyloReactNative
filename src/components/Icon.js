import React from 'react'
import { createIconSetFromFontello } from 'react-native-vector-icons'
import config from '../style/icon-config.json'
const Icon = createIconSetFromFontello(config)

export default function ({ src, size, color = '#000', styles, selected }) {
  return <Icon name={src} style={styles} {...{size, color, selected}} />
}
