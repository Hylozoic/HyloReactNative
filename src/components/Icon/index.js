import config from './config.json'
import { createIconSet } from 'react-native-vector-icons'

const glyphMap = config.icons.reduce((m, { tags, defaultCode }) => {
  m[tags[0]] = defaultCode
  return m
}, {})

export default createIconSet(glyphMap, 'hylo-evo-icons')
