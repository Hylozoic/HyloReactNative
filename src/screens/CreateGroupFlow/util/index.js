export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (groupDomain) {
  if (!groupDomain) return
  let formattedDomain = groupDomain.replace(/hylo\.com\/groups\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/groups/' + formattedDomain
  }
  return formattedDomain
}

export function removeDomainFromURL (groupDomain) {
  if (!groupDomain) return
  return groupDomain.replace('hylo.com/groups/', '')
}
