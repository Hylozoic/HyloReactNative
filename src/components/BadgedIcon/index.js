import React from 'react'
import { View } from 'react-native'
import Icon from '../Icon'
import styles from './BadgedIcon.styles'

export default function BadgedIcon (props) {
  return <View>
    <Icon { ...props } />
    {props.showBadge && <View style={styles.badgeOuter}>
      <View style={styles.badgeInner}></View>
    </View>}
  </View>
}
