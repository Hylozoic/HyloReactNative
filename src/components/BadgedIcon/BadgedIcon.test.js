import 'react-native'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import TestRenderer from 'react-test-renderer'

import BadgedIcon from './'
import Icon from '../Icon'
import { simulate } from 'util/testing'

describe('BadgedIcon', () => {
  it('matches the last snapshot without the badge', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<BadgedIcon name='Notifications' onPress={() => {}} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('matches the last snapshot with the badge', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<BadgedIcon name='Notifications' onPress={() => {}} showBadge />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('calls the onPress passed to the wrapped component', () => {
    const onPress = jest.fn()
    const root = TestRenderer.create(<BadgedIcon onPress={onPress} />).root
    simulate(root.findByType(Icon), 'press')
    expect(onPress).toHaveBeenCalled()
  })
})
