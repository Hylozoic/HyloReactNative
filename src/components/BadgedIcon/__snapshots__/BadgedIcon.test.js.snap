// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgedIcon matches the last snapshot with the badge 1`] = `
<View>
  <Icon
    name="Notifications"
    onPress={[Function]}
    showBadge={true}
  />
  <View>
    <View />
  </View>
</View>
`;

exports[`BadgedIcon matches the last snapshot without the badge 1`] = `
<View>
  <Icon
    name="Notifications"
    onPress={[Function]}
  />
</View>
`;
